Modules/Scripted/VPAWModel/VPAWModel.py

```diff
@@ -361,74 +381,395 @@ def onApplyButton(self):
+            and self.linkPediatricAirwayAtlas(pediatricAirwayAtlasDirectory)
+        ):
+            # We don't have self.pediatric_airway_atlas and we couldn't get it.
+            return False
 
         startTime = time.time()
-        logging.info("Processing started")
-
-        # Compute the thresholded output volume using the "Threshold Scalar Volume" CLI
-        # module
-        cliParams = {
-            "InputVolume": inputVolume.GetID(),
-            "OutputVolume": outputVolume.GetID(),
-            "ThresholdValue": imageThreshold,
-            "ThresholdType": "Above" if invert else "Below",
-        }
-        cliNode = slicer.cli.run(
-            slicer.modules.thresholdscalarvolume,
-            None,
-            cliParams,
-            wait_for_completion=True,
-            update_display=showResult,
+        logging.info("Pediatric Airway Atlas pipeline started")
+
+        # self.convertCTScansToNRRD(vPAWRootDirectory)
+        response = self.convertFCSVLandmarksToP3(
+            vPAWRootDirectory, patientPrefix
+        ) and self.runSegmentation(
+            vPAWRootDirectory, vPAWModelsDirectory, patientPrefix
         )
-        # We don't need the CLI module node anymore, remove it to not clutter the scene
-        # with it
-        slicer.mrmlScene.RemoveNode(cliNode)
+        if response:
+            slicer.util.infoDisplay("The pipeline has completed", "Pipeline ran")
 
         stopTime = time.time()
-        logging.info(f"Processing completed in {stopTime-startTime:.2f} seconds")
+        logging.info(
+            f"Pediatric Airway Atlas pipeline completed in {stopTime-startTime:.2f}"
+            + " seconds"
+        )
+        return response
+
+    def convertFCSVLandmarksToP3(self, vPAWRootDirectory, patientPrefix):
+        cwd = os.getcwd()
+        try:
+            os.chdir(self.pediatric_airway_atlas_directory)
+
+            images_dir = os.path.join(vPAWRootDirectory, "images")
+            input_landmarks_dir = os.path.join(vPAWRootDirectory, "landmarks")
+            output_landmarks_dir = os.path.join(
+                vPAWRootDirectory, "transformed_landmarks"
+            )
+            num_workers = 1
+            subject_prefix = patientPrefix
+            if subject_prefix is not None and subject_prefix != "":
+                try:
+                    slicer.util._executePythonModule(
+                        "conversion_utils.generate_pixel_space_landmarks",
+                        [
+                            f"--images_dir={images_dir}",
+                            f"--input_landmarks_dir={input_landmarks_dir}",
+                            f"--output_landmarks_dir={output_landmarks_dir}",
+                            f"--num_workers={num_workers}",
+                            f"--subject_prefix={subject_prefix}",
+                        ],
+                    )
+                except:
+                    slicer.util.errorDisplay(
```

Comment:
This error branch seems to capture the case where I put in a bad subject ID, i.e one for which there is no input image. The error message is then confusing in that case

The message could be updated to say "It could also be that you entered a subject prefix that does not exist" in order to cover that possibility

Or there can be a separate error handling for a bad subject ID

Reply:
So far, I have been able to maintain a clean separation between `VPAW` and `pediatric_airway_atlas` in that only the latter ever cares about the subdirectory organization of and file contents within the `pediatric_airway_atlas/`, `models/`, or `vpaw-root-dir/` directories.  (Well, `VPAW` does know the names and locations of the top-level `pediatric_airway_atlas` Python modules that it is calling.) Rather than have `VPAW` look inside, checking for the existence of a particular patient prefix by knowing where to look in `vpaw-root-dir/`, I'll simply modify the error message along the lines that you suggest.

Reply:
Actually, in my testing, a bad patient prefix does not trigger any error.  The subject_prefix-enabled pipeline happily processes all zero files that it finds with that prefix.  That is, it happily does nothing.  So, I'm not so sure that the modified error messages (just pushed) are a net win.

Reply:
Oh, hmm. In my test it did error out when I did that. Let me try again...

Reply:
Hmm, okay so if the pipeline has already processed all images then when I put a bogus subject ID I get the same result as you: it happily processes zero files. But if the pipeline has some unprocessed images in there, then a bogus subject ID results in the error message.

Reply:
Good sleuthing!  I'll go ahead and merge with these messages modified as you suggest.